github/resource_github_user_gpg_key.go

```diff
@@ -0,0 +1,75 @@
+package github
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGithubUserGpgKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubUserGpgKeyCreate,
+		Read:   resourceGithubUserGpgKeyRead,
+		Delete: resourceGithubUserGpgKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"armored_public_key": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"key_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGithubUserGpgKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+
+	pubKey := d.Get("armored_public_key").(string)
+
+	key, _, err := client.Users.CreateGPGKey(context.TODO(), pubKey)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(*key.ID, 10))
```

Comment:
Should we use `key.KeyID` here instead? Looking at GitHub docs it looks like `key.ID` is a database index-like sequential key, where as `KeyID` appears to be a unique id.

It also saves us the trouble of using `FormatInt` and error checking

Reply:
I guess that depends on what `GetGPGKey` uses... the docs make it look like it uses `key_id` but I can't imagine your tests passed if that's what they expected and you weren't doing that 🤔

Reply:
I've answered my own question. How bizarre 🤔